tommath_private.h

```diff
@@ -85,7 +85,8 @@ int func_name (mp_int * a, type b)                       \
      mp_zero(a);                                         \
      while (b != 0u) {                                   \
         a->dp[x++] = ((mp_digit)b & MP_MASK);            \
-        b >>= DIGIT_BIT;                                 \
+        if (CHAR_BIT * sizeof (b) <= DIGIT_BIT) { break; } \
```

Comment:
I was just writing this comment while you pushed the change...

my question is if it is really `CHAR_BIT` as suggested by @czurnieden or if 8 is correct...

I think of platforms like some DSP's where `CHAR_BIT` is 16 or 32

I guess the 8 must stay, but I'm not 100% sure yet and I've to leave...

Reply:
Ok, probably. Since sizeof always counts bytes??

Reply:
But we have sizeof*CHAR_BIT at other places too. Then we would have to remove all of those and replace with 8.

Reply:
According to stackoverflow, sizeof is always bytes. https://stackoverflow.com/questions/11868211/does-sizeof-return-the-number-of-bytes-or-the-number-of-octets-of-a-type-in-c

Therefore the usage of CHAR_BIT is wrong at many places in tommath. I can fix this in a separate PR if you wish.

Reply:
Ah and bytes are implementation defined. Hehe. This is shit.

Reply:
@sjaeckel Please merge as is and we resolve the CHAR_BIT question in a separate PR. If we want to resolve it...

Reply:
> Ah and bytes are implementation defined. Hehe. This is shit.

Oh, yes! \*deep sigh\*

To quote the standard (ISO/IEC 9899:2011 (E). I'm still too stingy to get the new one)

<blockquote>
  3.6<br>
1 byte<br>
  addressable unit of data storage large enough to hold any member of the basic character
  set of the execution environment<br>
2 NOTE 1  It is possible to express the address of each individual byte of an object uniquely.<br>
3 NOTE 2 A byte is composed of a contiguous sequence of bits, <b>the number of which is implementation-defined</b>. The least significant bit is called the low-order bit; the most significant bit is called the high-order bit.
</blockquote> (emphasis by me)

But also
<blockquote>
6.5.3.4 The sizeof and _Alignof operators<br>
&hellip;<br>
2 The sizeof operator yields the size (in bytes) of its operand, [&hellip;]<br>
4 When sizeof is applied to an operand that has type char, unsigned char, or
  signed char, (or a qualified version thereof) the result is 1.
</blockquote>

So you can get the size of a byte with all standard compliant compilers where it is in `CHAR_BIT`. But what is with all of the non-standard compliant compilers? They are not supported, you said? Well, we really need a list of supported ones somewhere in a place where we can point at it if somebody wants to blame us for the problems one of the many crappy c-compilers out there gave them.

Reply:
Ok at least CHAR_BIT=BYTES_BIT. Then it is correct as it is now :)